apps/backend/src/plugins/authorization/authorization.plugin.ts

```diff
@@ -0,0 +1,90 @@
+import { type FastifyRequest } from "fastify";
+import fp from "fastify-plugin";
+import { type JWTPayload, jwtVerify } from "jose";
+import micromatch from "micromatch";
+
+import { config } from "~/libs/modules/config/config.js";
+import { type UserService } from "~/modules/users/user.service.js";
+
+import {
+	AuthorizationError,
+	ErrorMessage,
+	type UserSignUpResponseDto,
+} from "./libs/types/types.js";
+
+declare module "fastify" {
+	interface FastifyInstance {
+		authenticate: (request: FastifyRequest) => Promise<void>;
+	}
+
+	interface FastifyRequest {
+		user: UserSignUpResponseDto;
+	}
+}
+
+type AuthPluginOptions = {
+	userService: UserService;
+	whiteRoutes: string[];
+};
+
+const extractUserFromRequest = async (
+	request: FastifyRequest,
+	userService: UserService,
+): Promise<UserSignUpResponseDto> => {
+	try {
+		const { authorization } = request.headers;
+
+		if (!authorization?.startsWith("Bearer ")) {
```

Comment:
Please, move "Bearer" to a separate constant and reuse everywhere where needed this constant instead of a string